sim/dex-data.ts

```diff
@@ -102,14 +116,12 @@ export class BasicEffect implements EffectData {
 	sourceEffect: string;
 
 	constructor(data: AnyObject) {
-		this.exists = true;
-		Object.assign(this, data);
-
 		this.name = Utils.getString(data.name).trim();
 		this.id = data.realMove ? toID(data.realMove) : toID(this.name); // Hidden Power hack
 		this.fullname = Utils.getString(data.fullname) || this.name;
 		this.effectType = Utils.getString(data.effectType) as EffectType || 'Condition';
-		this.exists = !!(this.exists && this.id);
+		const exists = Object.prototype.propertyIsEnumerable.call(data, 'exists') ? data.exists : true;
```

Comment:
This is only written this way in an effort to preserve the exact semantics of the old code.

It's up to you if you want to replace it with simpler code that has slightly different semantics.